service/tablet_allocator.cc

```diff
@@ -2443,15 +2443,27 @@ class load_balancer {
                     lblogger.info("Will drain node {} ({}) from DC {}", node.host_id(), node.get_state(), dc);
                     nodes_to_drain.emplace(node.host_id());
                     nodes[node.host_id()].drained = true;
-                } else if (node.is_excluded() || _skiplist.contains(node.host_id())) {
+                } else if (node.is_excluded()) {
                     // Excluded nodes should not be chosen as targets for migration.
-                    lblogger.debug("Ignoring excluded or dead node {}: state={}", node.host_id(), node.get_state());
+                    lblogger.debug("Ignoring excluded node {}: state={}", node.host_id(), node.get_state());
                 } else {
                     ensure_node(node.host_id());
                 }
             }
         });
 
+        // Apply skiplist only when not draining.
+        // It's unsafe to move tablets to non-skip nodes as this can lead to node overload.
+        if (nodes_to_drain.empty()) {
+            for (auto h : _skiplist) {
+                if (nodes.contains(h)) {
```

Comment:
It'd be more efficient to use [extract](https://en.cppreference.com/w/cpp/container/unordered_map/extract) instead, since it will lookup the key once and allow you to access the node& before it's erased.

Reply:
Otherwise LGTM

Reply:
Ok.